src/nimble_mpi.cc

```diff
@@ -313,6 +315,21 @@ int NimbleMain(std::shared_ptr<MaterialFactoryType> material_factory,
   int contact_force_field_id = model_data.AllocateNodeData(nimble::VECTOR, "contact_force", num_nodes);
   int skin_node_field_id = model_data.AllocateNodeData(nimble::SCALAR, "skin_node", num_nodes);
 
+  //
```

Comment:
Same comment here: the type of the factory created needs to be prescribed through some kind of (preferably abstract) interface for compatibility with Extras.